pkg/agent/client.go

```diff
@@ -404,13 +404,22 @@ func (a *Client) Serve() {
 				<-dialDone
 				if connCtx.conn != nil {
 					klog.V(4).InfoS("close connection", "connectionID", connID)
-					closeResp := &client.Packet{
-						Type:    client.PacketType_CLOSE_RSP,
-						Payload: &client.Packet_CloseResponse{CloseResponse: &client.CloseResponse{}},
+					var closePkt *client.Packet
+					if connID == 0 {
```

Comment:
isn't this impossible? Given line 394 I would expect connID to always be at least 1.

Reply:
Good point. The intent here is that if the dial response hasn't been sent, then the server isn't aware of the connection ID. In general, our close semantics are leaky. We really need something like a `DIAL_ACK` to confirm that the connection ID was "registered". Better yet, if we got rid of the dialID / connectionID separation and just had a single connection ID initiated at the get go than we could simplify a lot of this... Unfortunately no one seems to remember why they were separated in the first place.